demos/projects/ESPRESSIF/esp32/components/azure-iot-middleware-freertos/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+set(ROOT_PATH
+    ${CMAKE_CURRENT_LIST_DIR}/../../../../../..
+)
+
+set(AZURE_IOT_MIDDLEWARE_FREERTOS 
+    ${ROOT_PATH}/libs/azure-iot-middleware-freertos
+)
+
+file(GLOB_RECURSE COMPONENT_SOURCES
+    ${AZURE_IOT_MIDDLEWARE_FREERTOS}/source/*.c  
+)
+
+list(APPEND COMPONENT_SOURCES
+    ${AZURE_IOT_MIDDLEWARE_FREERTOS}/ports/coreMQTT/azure_iot_core_mqtt.c
+)
+
+idf_component_get_property(FREERTOS_DIR freertos COMPONENT_DIR)
+
+set(COMPONENT_INCLUDE_DIRS
+    ${FREERTOS_DIR}/include/freertos
+    ${AZURE_IOT_MIDDLEWARE_FREERTOS}/.github/config
+    ${AZURE_IOT_MIDDLEWARE_FREERTOS}/source/include
+    ${AZURE_IOT_MIDDLEWARE_FREERTOS}/source/interface
+    ${AZURE_IOT_MIDDLEWARE_FREERTOS}/ports/coreMQTT
+)
+
+idf_component_register(
+    SRCS ${COMPONENT_SOURCES}
+    INCLUDE_DIRS ${COMPONENT_INCLUDE_DIRS}
+    REQUIRES freertos azure-sdk-for-c coreMQTT)
```

Comment:
>azure-sdk-for-c [](http://example.com/codeflow?start=22&length=15)

how about making middleware as component ? In that case middleware will bring in middleware and embedded sdk. For end user middleware is sdk to connect to azure iot.

Reply:
You have to have azure-sdk-for-c as a component that is Required by the middleware component for the ESP Build system. But this should use the dependent build model for components not the requires here.

Reply:
As we spoke, we decided to keep the components separate.